src/aihawk_authenticator.py

```diff
@@ -1,22 +1,34 @@
 import random
 import time
 
-from selenium.common.exceptions import NoSuchElementException, TimeoutException, NoAlertPresentException, TimeoutException, UnexpectedAlertPresentException
+from loguru import logger
+from selenium.common.exceptions import NoSuchElementException, TimeoutException
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.support.ui import WebDriverWait
 
-from loguru import logger
-
 
 class AIHawkAuthenticator:
 
-    def __init__(self, driver=None):
+    def __init__(self, driver=None, bot_facade=None):
         self.driver = driver
+        self.email = bot_facade.email if bot_facade else ""
+        self.password = bot_facade.password if bot_facade else ""
         logger.debug(f"AIHawkAuthenticator initialized with driver: {driver}")
 
+    def set_bot_facade(self, bot_facade):
+        self.email = bot_facade.email
+        self.password = bot_facade.password
+        logger.debug(f"Email and password set from bot_facade: email={self.email}, password={'*' * len(self.password)}")
+
+
     def start(self):
-        logger.info("Starting Chrome browser to log in to AIHawk.")
+        logger.info("Starting Chrome browser to log in to LinkedIn.")
```

Comment:
this log could be misleading,  as the browser depends on the engine. Wouldn't it be better to change to browser?

Reply:
can we use wrapper so it can be run on debug or info mode

```
class LoggerWrapper:
    def __init__(self, logger):
        self.logger = logger
        self.debug_mode = os.environ.get('DEBUG', 'False').lower() == 'true'

    def debug(self, message, *args, **kwargs):
        if self.debug_mode:
            self.logger.debug(message, *args, **kwargs)

    def info(self, message, *args, **kwargs):
        self.logger.info(message, *args, **kwargs)

    def __getattr__(self, name):
        return getattr(self.logger, name)

```